src/main/java/org/osiam/ResourceServer.java

```diff
@@ -0,0 +1,220 @@
+import org.springframework.transaction.annotation.*;
+import org.springframework.web.filter.*;
+import org.springframework.web.servlet.config.annotation.*;
+
+import com.codahale.metrics.*;
+import com.fasterxml.jackson.databind.*;
+import com.zaxxer.hikari.*;
+
+@Configuration
+@ComponentScan
+@EnableWebMvc
+@EnableWebSecurity
+@EnableTransactionManagement
+@EnableAspectJAutoProxy(proxyTargetClass = true)
+public class ResourceServer {
+
+    @Value("${org.osiam.resource-server.db.driver}")
+    private String driverClassName;
+
+    @Value("${org.osiam.resource-server.db.url}")
+    private String databaseUrl;
+
+    @Value("${org.osiam.resource-server.db.username}")
+    private String databaseUserName;
+
+    @Value("${org.osiam.resource-server.db.password}")
+    private String databasePassword;
+
+    @Value("${org.osiam.resource-server.db.dialect}")
+    private String databasePlatform;
+
+    @Bean
+    public static PropertySourcesPlaceholderConfigurer properties() {
+        PropertySourcesPlaceholderConfigurer propertySources = new PropertySourcesPlaceholderConfigurer();
+        Resource[] resources = new ClassPathResource[] { new ClassPathResource("resource-server.properties") };
+        propertySources.setLocations(resources);
+        propertySources.setIgnoreUnresolvablePlaceholders(true);
+        return propertySources;
+    }
+
+    @Bean
+    public Filter characterEncodingFilter() {
+        CharacterEncodingFilter characterEncodingFilter = new CharacterEncodingFilter();
+        characterEncodingFilter.setEncoding("UTF-8");
+        characterEncodingFilter.setForceEncoding(true);
+        return characterEncodingFilter;
+    }
+
+    @Bean
+    public Filter createSSLFilter() {
+        return new SSLRequestLoggingFilter();
+    }
+
+    @Bean
+    @Primary
+    public ObjectMapper objectMapper() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        return objectMapper;
+    }
+
+    @Primary
+    @Bean(destroyMethod = "close")
+    public DataSource dataSource() {
+        return new HikariDataSource(hikariConfig());
+    }
+
+    @Bean(initMethod = "migrate")
+    @Autowired
+    public Flyway flyway(DataSource dataSource,
+            @Value("${org.osiam.resource-server.db.vendor:postgresql}") String databaseVendor) {
+        Flyway flyway = new Flyway();
+        flyway.setBaselineOnMigrate(true);
+        flyway.setDataSource(dataSource);
+        flyway.setLocations("db/migration/" + databaseVendor);
+        flyway.setTable("resource_server_schema_version");
+        MigrationVersion version = MigrationVersion.fromVersion("0");
+        flyway.setBaselineVersion(version);
+        return flyway;
+    }
+
+    @Bean
+    public HikariConfig hikariConfig() {
+        HikariConfig hikariConfig = new HikariConfig();
+        hikariConfig.setPoolName("osiam-auth-server-cp");
```

Comment:
Yes, the resource-server shares the connection pool with the auth-server :D No this is a copy waste error. The correct pool name is `osiam-resource-server-cp`.